enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/context.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  createContext,
+  type PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { useUnmount } from "react-use";
+
+import type { SdkPluginsConfig } from "embedding-sdk";
+import { getDefaultVizHeight } from "embedding-sdk/lib/default-height";
+import { useSdkSelector } from "embedding-sdk/store";
+import { getPlugins } from "embedding-sdk/store/selectors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { initializeQBRaw, resetQB } from "metabase/query_builder/actions";
+import {
+  getCard,
+  getFirstQueryResult,
+  getQueryResults,
+  getQuestion,
+  getUiControls,
+} from "metabase/query_builder/selectors";
+import type { Mode } from "metabase/visualizations/click-actions/Mode";
+import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+import type Question from "metabase-lib/v1/Question";
+import type { Card, Dataset } from "metabase-types/api";
+import type { QueryBuilderUIControls } from "metabase-types/store";
+
+type InteractiveQuestionContextType = {
+  question: Question | undefined;
+  card: Card | null;
+  result: Dataset | null;
+  uiControls: QueryBuilderUIControls;
+  queryResults: Dataset[] | null;
+  plugins: SdkPluginsConfig | null;
+  mode: Mode | null | undefined;
+  defaultHeight?: number;
+  isQuestionLoading: boolean;
+  isQueryRunning: boolean;
+  resetQuestion: () => void;
+  onReset?: () => void;
+  onNavigateBack?: () => void;
+  withTitle?: boolean;
+  customTitle?: React.ReactNode;
+  withResetButton?: boolean;
```

Comment:
I think not all of these has to be in the context, as many of these can be queried externally e.g. with `useSelector`, but I'll leave this to other reviewers to share your feedback.

Reply:
I'll create a separate PR to have this discussion, as there are quite a few PRs depending on this one. I'll ping everybody once that happens.